server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -229,6 +230,7 @@ private void startNewSnapshots(SnapshotsInProgress snapshotsInProgress) {
         final String localNodeId = clusterService.localNode().getId();
         for (SnapshotsInProgress.Entry entry : snapshotsInProgress.entries()) {
             final State entryState = entry.state();
+            final Version version = SnapshotsService.compatibleVersion(entry);
```

Comment:
let's use and propagate the boolean here, not this version.

Reply:
Done